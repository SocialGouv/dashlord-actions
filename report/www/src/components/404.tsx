import * as React from "react";

import Table from "@codegouvfr/react-dsfr/Table";

import { Panel } from "./Panel";

type Report404Props = { data: Wget404Report };

export const Report404: React.FC<Report404Props> = ({ data }) => {
  const tableData = data && [
    ["destination"],
    ...data.map((url2) => [url2.link]),
  ];
  return (
    (data && data.length && (
      <Panel title="Erreurs 404" info="Pages introuvables">
        <Table data={tableData} />
      </Panel>
    )) ||
    null
  );
};
