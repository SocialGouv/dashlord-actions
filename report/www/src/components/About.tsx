import * as React from "react";

import CallOut from "@codegouvfr/react-dsfr/CallOut";

export const About: React.FC = () => (
  <CallOut className="fr-mb-3w" title="À propos">
    DashLord est <strong>100% open source</strong>, vous pouvez y contribuer en
    créant de la documentation, en proposant du code, ou encore en partageant
    vos questions{" "}
    <a
      href="https://github.com/socialgouv/dashlord"
      target="_blank"
      rel="noopener noreferrer"
    >
      sur le repository github
    </a>
    .
    <br /> <br />
    L&apos;objectif de DashLord est d&apos;identifier des points
    d&apos;attention mais aussi de proposer des solutions actionnables.
    <br /> <br />
    DashLord est né à la{" "}
    <a
      href="https://fabrique.social.gouv.fr"
      target="_blank"
      rel="noopener noreferrer"
    >
      {" "}
      Fabrique des ministères sociaux
    </a>{" "}
    pour répondre aux besoins d&apos;évaluation et de mise en oeuvre des bonnes
    pratiques de développement web.
    <br /> <br />
    Vous pouvez facilement créer votre propre DashLord pour vos URLs en suivant{" "}
    <a
      href="https://github.com/SocialGouv/dashlord/blob/main/README.md"
      target="_blank"
      rel="noopener noreferrer"
    >
      les instructions sur le GitHub
    </a>
    .
  </CallOut>
);
