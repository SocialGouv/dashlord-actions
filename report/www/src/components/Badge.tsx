import classNames from "classnames";
import React, { CSSProperties } from "react";
import { useRouter } from "next/router";

import styles from "./badge.module.scss";

type BadgeProps = {
  variant: string;
  className?: string;
  to?: string;
  style?: CSSProperties;
  external?: boolean;
  children: any;
};

const Badge: React.FC<BadgeProps> = ({
  children,
  external = false,
  variant,
  className,
  to,
  style,
}) => {
  const router = useRouter();
  return (
    <button
      type="button"
      className={classNames(className, styles[variant])}
      onClick={() => {
        if (to) {
          if (external) {
            window.open(to);
            return;
          }
          router.push(to);
        }
      }}
      style={style}
    >
      {children}
    </button>
  );
};

export default Badge;
