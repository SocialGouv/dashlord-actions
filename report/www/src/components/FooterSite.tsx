import React from "react";

import { Footer } from "@codegouvfr/react-dsfr/Footer";
import { headerFooterDisplayItem } from "@codegouvfr/react-dsfr/Display";

import dashlordConfig from "@/config.json";

export const FooterSite: React.FC = () => {
  //const Logo = dashlordConfig.marianne ? Marianne : () => <div />;
  return (
    <Footer
      classes={{
        logo: (!!dashlordConfig.marianne && "auto") || "hidden",
      }}
      brandTop={dashlordConfig.entity}
      homeLinkProps={{
        href: "/",
        title: "Accueil",
      }}
      operatorLogo={
        dashlordConfig.operator &&
        (typeof dashlordConfig.operator.logo === "string"
          ? {
              orientation: "horizontal",
              imgUrl: dashlordConfig.operator.logo,
              alt: `Logo ${dashlordConfig.operator.name}`,
            }
          : {
              orientation: dashlordConfig.operator.logo.direction,
              imgUrl: dashlordConfig.operator.logo.src,
              alt: `Logo ${dashlordConfig.operator.name}`,
            })
      }
      accessibility="non compliant"
      bottomItems={[headerFooterDisplayItem]}
      contentDescription={dashlordConfig.footer}
      license={dashlordConfig.marianne ? undefined : false}
    />
  );

  /*
    <Footer>
      <FooterTop></FooterTop>
      <FooterBody description="">
        <Logo
          style={{
            display: dashlordConfig.marianne === true ? "block" : "none",
          }}
        >
          {dashlordConfig.entity}
        </Logo>
        {dashlordConfig.operator && (
          <FooterOperator>
            { typeof dashlordConfig.operator.logo === "string" ? (
              <img className="fr-footer__logo" style={{width: "3.5rem"}} src={dashlordConfig.operator.logo} alt={dashlordConfig.operator.name} />
            ) : (
              <img className="fr-footer__logo" style={dashlordConfig.operator.logo.direction === "vertical" ? { maxWidth: "9.0625rem" } : {width: "3.5rem"}} src={dashlordConfig.operator.logo.src} alt={dashlordConfig.operator.name} />
            )}
          </FooterOperator>
        )}
        <FooterBodyItem>
          <div>{dashlordConfig.footer}</div>
        </FooterBodyItem>
      </FooterBody>
      <FooterBottom>
        <FooterLink href="#"></FooterLink>
        <FooterCopy href="#"></FooterCopy>
      </FooterBottom>
    </Footer>
  */
};
