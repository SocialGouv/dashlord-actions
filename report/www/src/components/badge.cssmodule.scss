.success,
.info,
.warning,
.danger {
    display: inline-block;
    cursor: pointer;
    padding: 4px 8px;
    color: white;
    min-width: 60px;
    border-radius: 4px;
    text-align: center;
    font-weight: bold;
}

.success {
    background-color: var(--success);
}

.info {
    background-color: var(--info);
}

.warning {
    background-color: #FF6F4C;
}

.danger {
    background-color: var(--error);
}