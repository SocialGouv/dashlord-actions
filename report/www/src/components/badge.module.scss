.success,
.info,
.warning,
.danger {
    display: inline-block;
    cursor: pointer;
    padding: 4px 8px;
    color: white;
    min-width: 60px;
    border-radius: 4px;
    text-align: center;
    font-weight: bold;
}



.success, .success:hover {
    background-color: var(--success) !important;
}

.info, .info:hover {
    background-color: var(--info) !important;
}

.warning, .warning:hover {
    background-color: var(--warning) !important;
}

.danger, .danger:hover {
    background-color: var(--error) !important;
}