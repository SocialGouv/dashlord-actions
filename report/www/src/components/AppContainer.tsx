import * as React from "react";

import { Container, Alert } from "@dataesr/react-dsfr";

import { Switch, Route, useLocation, useParams } from "react-router-dom";

import { Dashboard } from "./Dashboard";
import { Trends } from "./Trends";
import { Intro } from "./Intro";
import { About } from "./About";
import { WappalyzerDashboard } from "./WappalyzerDashboard";
import Url from "./Url";

const report: DashLordReport = require("./report.json");
const trends: Trends = require("./trends.json");

type CategoryRouteProps = { report: DashLordReport };

// for some reason react-router `:url*` didnt work, use `*` only
interface CategoryParamTypes {
  category: string;
}

const CategoryRoute: React.FC<CategoryRouteProps> = (props) => {
  const params = useParams<CategoryParamTypes>();
  const category = window.decodeURIComponent(params.category);
  const urls = props.report.filter((u) => u.category === category);
  return (
    <>
      <br />
      {urls.length ? (
        <>
          <h3>
            {category} :{urls.length} urls
          </h3>
          <Dashboard report={urls} />
        </>
      ) : (
        <h3>Aucun URL associ√©e</h3>
      )}
    </>
  );
};

interface TagParamTypes {
  tag: string;
}

type TagRouteProps = { report: DashLordReport };

const TagRoute: React.FC<TagRouteProps> = (props) => {
  const params = useParams<TagParamTypes>();
  const tag = window.decodeURIComponent(params.tag);
  const urls = props.report.filter((u) => u.tags && u.tags.includes(tag));
  return (
    <>
      <br />
      <h3>
        {tag} :{urls.length} urls
      </h3>
      <Dashboard report={urls} />
    </>
  );
};

interface UrlParamTypes {
  "0": string;
}

type UrlRouteProps = { report: DashLordReport };

const UrlRoute: React.FC<UrlRouteProps> = (props) => {
  const params = useParams<UrlParamTypes>();
  const url = window.decodeURIComponent(params["0"]);
  const urlData = props.report.find((u) => u.url === url);
  if (!urlData) {
    return (
      <Alert
        type="error"
        title={`Impossible de trouver le rapport pour ${url}`}
      />
    );
  }
  return <Url url={url} report={urlData} />;
};

export const AppContainer = () => {
  const location = useLocation();
  return (
    <Container fluid={location.pathname === "/"}>
      <div role="main" className="fr-my-4w">
        <Switch>
          <Route path="/url/*">
            <UrlRoute report={report} />
          </Route>
          <Route path="/dashboard">
            <Dashboard report={report} />
          </Route>
          <Route path="/trends">
            <Trends trends={trends} />
          </Route>
          <Route path="/category/:category">
            <CategoryRoute report={report} />
          </Route>
          <Route path="/tag/:tag">
            <TagRoute report={report} />
          </Route>
          <Route path="/wappalyzer">
            <WappalyzerDashboard report={report} />
          </Route>
          <Route path="/intro">
            <Intro />
          </Route>
          <Route path="/about">
            <About />
          </Route>
          <Route path="/">
            <Dashboard report={report} />
          </Route>
        </Switch>
      </div>
    </Container>
  );
};
