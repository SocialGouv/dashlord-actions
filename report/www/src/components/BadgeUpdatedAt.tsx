import { formatDistanceToNow, differenceInDays, format } from "date-fns";
import { fr } from "@codegouvfr/react-dsfr";
import frLocale from "date-fns/locale/fr";

import { AlertProps } from "@codegouvfr/react-dsfr/Alert";
import Badge from "@codegouvfr/react-dsfr/Badge";
import { CSSProperties } from "react";

// badge that displya "since xxx" and appropriate colors
export const BadgeUpdatedAt = ({
  date,
  style,
}: {
  date: string;
  style: CSSProperties;
}) => {
  if (!date) return;
  const startDate = new Date(date);
  const lagInDays = differenceInDays(new Date(), startDate);
  let severity: AlertProps.Severity = "info";
  if (lagInDays > 60) {
    severity = "error";
  } else if (lagInDays > 30) {
    severity = "warning";
  }
  return (
    <Badge
      severity={severity}
      noIcon
      as="span"
      style={{ verticalAlign: "top", ...style }}
    >
      <i
        className={fr.cx("fr-icon-time-fill", "fr-icon--sm", "fr-mr-1v")}
        title={`Mis Ã  jour le ${format(new Date(startDate), "dd/M/yyyy")}`}
      />
      {formatDistanceToNow(new Date(startDate), {
        locale: frLocale,
      })}
    </Badge>
  );
};
