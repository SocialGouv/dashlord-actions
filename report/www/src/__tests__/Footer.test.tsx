import { render } from "@testing-library/react";
import { FooterSite } from "@/components/FooterSite";
import { vi } from "vitest";

it("Should render FooterSite", () => {
  const { container } = render(<FooterSite />);
  expect(container).toMatchSnapshot();
});
