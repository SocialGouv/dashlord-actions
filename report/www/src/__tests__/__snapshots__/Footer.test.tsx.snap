// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Should render FooterSite 1`] = `
<div>
  <footer
    class="fr-footer"
    id="fr-footer"
    role="contentinfo"
  >
    <div
      class="fr-container"
    >
      <div
        class="fr-footer__body"
      >
        <div
          class="fr-footer__brand fr-enlarge-link"
        >
          <p
            class="fr-logo auto"
          >
            Ministères sociaux
          </p>
          <a
            class="fr-footer__brand-link"
            href="/"
            title="Accueil"
          >
            <img
              alt="Logo Fabrique des ministères sociaux"
              class="fr-footer__logo"
              src="https://avatars.githubusercontent.com/u/35456799?s=400"
              style="max-width: 9.0625rem;"
            />
          </a>
        </div>
        <div
          class="fr-footer__content"
        >
          <p
            class="fr-footer__content-desc"
          >
            Un produit de la fabrique numérique des ministères sociaux, de la communauté beta.gouv.fr
          </p>
          <ul
            class="fr-footer__content-list"
          >
            <li
              class="fr-footer__content-item"
            >
              <a
                class="fr-footer__content-link"
                href="https://legifrance.gouv.fr"
                target="_blank"
                title="legifrance.gouv.fr - ouvre une nouvelle fenêtre"
              >
                legifrance.gouv.fr
              </a>
            </li>
            <li
              class="fr-footer__content-item"
            >
              <a
                class="fr-footer__content-link"
                href="https://gouvernement.fr"
                target="_blank"
                title="gouvernement.fr - ouvre une nouvelle fenêtre"
              >
                gouvernement.fr
              </a>
            </li>
            <li
              class="fr-footer__content-item"
            >
              <a
                class="fr-footer__content-link"
                href="https://service-public.fr"
                target="_blank"
                title="service-public.fr - ouvre une nouvelle fenêtre"
              >
                service-public.fr
              </a>
            </li>
            <li
              class="fr-footer__content-item"
            >
              <a
                class="fr-footer__content-link"
                href="https://data.gouv.fr"
                target="_blank"
                title="data.gouv.fr - ouvre une nouvelle fenêtre"
              >
                data.gouv.fr
              </a>
            </li>
          </ul>
        </div>
      </div>
      <div
        class="fr-footer__bottom"
      >
        <ul
          class="fr-footer__bottom-list"
        >
          <li
            class="fr-footer__bottom-item"
          >
            <span
              class="fr-footer__bottom-link"
            >
              Accessibilité: non conforme
            </span>
          </li>
          <li
            class="fr-footer__bottom-item"
          >
            <button
              aria-controls="fr-theme-modal"
              class="fr-footer__bottom-link fr-icon-theme-fill fr-link--icon-left"
              data-fr-opened="false"
              id="fr-theme-modal-control-button"
            >
              Paramètres d'affichage
            </button>
          </li>
        </ul>
        <div
          class="fr-footer__bottom-copy"
        >
          <p>
            Sauf mention explicite de propriété intellectuelle détenue par des tiers, les contenus de ce site sont proposés sous
             
            <a
              href="https://github.com/etalab/licence-ouverte/blob/master/LO.md"
              target="_blank"
              title="licence etalab-2.0 - ouvre une nouvelle fenêtre"
            >
              licence etalab-2.0
            </a>
          </p>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
